test/language/number/scientific_literals.wren

```diff
@@ -0,0 +1,9 @@
+var x = 2.55e2
+
+IO.print(x) // expect: 255
+IO.print(x + 1) // expect: 256
+IO.print(x == 255) // expect: true
+IO.print(2.55e-2 is Num) // expect: true
+IO.print(x is Num) // expect: true
+IO.print(-2.55e2) // expect: -255
+IO.print(-25500e-2) // expect: -255
```

Comment:
For the very long exponent and mantissa, I don't really know how to write the tests, since it is the resulting number that counts. If `9999....9999` results in "too large literal", adding `e-100` to it will no longer trigger the error.